Parser.cpp

```diff
@@ -0,0 +1,40 @@
+#include "Parser.h"
+
+const std::map <std::string, std::string> Parser::parseJson(std::string json) {
+    static const std::regex parseRegex("\\s*\"([\\w]*)\"\\s*:\\s*\"?([\\w\\.]*)\"?\\s*[,}]\\s*");
+    std::smatch matches;
+    std::map<std::string,std::string> attributes;
+
+    static const std::regex fileNameRegex("([\\w]*).json$");
+    if (std::regex_search(json, matches, fileNameRegex))
+    {
+        std::ifstream jsonFile;
+        jsonFile.open(json);
+        if (jsonFile.fail()) throw json + " does not exist!";
+        else
+        {
+           attributes = parseJson(jsonFile);
+           jsonFile.close();
+           return attributes;
+        }
+    }
```

Comment:
Ne keverd össze a két funkciót! Most ugyanaz a fv. próbálja meg egy fájlnév alapján elvégezni a parzolást, és egy string tartalma alapján is.

Reply:
A stringes lehetőséget átraktam egy külön függvénybe.